packages/next/src/server/stream-utils/node-web-streams-helper.ts

```diff
@@ -639,7 +647,13 @@ export async function continueFizzStream(
 
   // If we're generating static HTML and there's an `allReady` promise on the
   // stream, we need to wait for it to resolve before continuing.
-  if (isStaticGeneration && 'allReady' in renderStream) {
+  // However, if allowStreamingDuringStaticGeneration is true, we should allow
+  // streaming even during static generation (useful for force-static pages).
+  if (
+    isStaticGeneration &&
+    !allowStreamingDuringStaticGeneration &&
+    'allReady' in renderStream
+  ) {
     await renderStream.allReady
```

Comment:
Thanks for looking into this (I created the main issue)

Question: I understand the need to wait for static site to generate: because you want the HTML without suspense to end up in the cache (not the one streamed to the client)

So conceptually I think Next needs to (i) send the streamed version to the client (ii) while also generating the HTML (with no suspense boundaries) to make that end up in the cache

Otherwise having the suspense version in the cache seems not ideal as it might cause unnecessary jumps (data is already all there)